torchaudio/datasets/esc.py

```diff
@@ -0,0 +1,185 @@
+import os
+import torchaudio
+from torch import Tensor
+from torch.nn import Module
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import download_url, extract_archive
+import numpy
+from typing import Tuple
+
+# Download URL and checksums
+URL = {
+    "esc-10": "https://github.com/karoldvl/ESC-50/archive/master.zip",
+    "esc-50": "https://github.com/karoldvl/ESC-50/archive/master.zip",
+    # "esc-us": None,
+}
+
+_CHECKSUMS = {
+    "esc-10": None,
+    "esc-50": None,
+    # "esc-us": None,
+}
+
+# Constant
+ARCHIVE_BASENAME = "ESC-50-master"
+FOLDER_IN_ARCHIVE = "ESC-50-master"
+AUDIO_FOLDER = "audio"
+META_FOLDER = "meta"
+AVAILABLE_VERSION = list(URL.keys())
+
+# Default parameters
+FOLDS = (1, 2, 3, 4, 5)
+
+
+class ESC50(Dataset):
+    """
+    ESC datasets
+
+    Args:
+        root (string): Root directory of datasets where directory
+            ``ESC-50-master`` exists or will be saved to if download is set to True.
+        download (bool, optional): If true, download the dataset from the internet
+            and puts it in root directory. If datasets is already downloaded, it is
+            not downloaded again.
+    """
+    NB_CLASS = 50
+
+    def __init__(self,
+                 root: str,
+                 folds: tuple = FOLDS,
+                 download: bool = False,
+                 transform: Module = None) -> None:
+
+        super().__init__()
+
+        self.root = root
+        self.required_folds = folds
+        self.transform = transform
+
+        self.url = URL["esc-50"]
+        self.nb_class = 50
```

Comment:
i'm assuming this is meant to be a property of the dataset to be read to the user? let's move that to the documentation for now?